pgosupport.cmake

```diff
@@ -19,25 +28,57 @@ function(add_pgo TargetName)
     )
 
     # Enable PGO only for optimized configs
-    set(ConfigTypeList RELEASE RELWITHDEBINFO)
+    set(ConfigList RELEASE RELWITHDEBINFO)
+    set(IsReleaseConfig "$<OR:$<CONFIG:RELEASE>,$<CONFIG:RELWITHDEBINFO>>")
 
-    foreach(ConfigType IN LISTS ConfigTypeList)
-        set(LinkFlagsProperty "LINK_FLAGS_${ConfigType}")
-        if(CLR_CMAKE_PGO_INSTRUMENT)
+    if(CLR_CMAKE_PGO_INSTRUMENT)
+        # Unfortunately LINK_FLAGS_* don't support generator expressions, so we need to use a loop
+        foreach(Config IN LISTS ConfigList)
             if(WIN32)
-                set_property(TARGET ${TargetName} APPEND_STRING PROPERTY ${LinkFlagsProperty} "/LTCG /GENPROFILE")
+                append_prop_strings(${TargetName} LINK_FLAGS_${Config} /LTCG /GENPROFILE)
```

Comment:
I can remove it, but I've already been setting it (on Windows) and it seems harmless to repeat it (PGO doesn't work without LTCG). Do we prefer for /LTCG to be centralized and a hard-fail if missed?